providers/fcm/src/lib/fcm.provider.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IPushOptions,
+  IPushProvider,
+} from '@novu/stateless';
+import { initializeApp, cert } from 'firebase-admin/app';
+import { getMessaging, Messaging } from 'firebase-admin/messaging';
+import crypto from 'crypto';
+
+export class FcmPushProvider implements IPushProvider {
+  id = 'fcm';
+  channelType = ChannelTypeEnum.PUSH as ChannelTypeEnum.PUSH;
+
+  private messaging: Messaging;
+  constructor(
+    private config: {
+      projectId: string;
+      email: string;
+      secretKey: string;
+    }
+  ) {
+    this.config = config;
+    const firebase = initializeApp(
+      {
+        credential: cert({
+          projectId: this.config.projectId,
+          clientEmail: this.config.email,
+          privateKey: JSON.parse(`"${this.config.secretKey}"`),
+        }),
+      },
+      crypto.randomBytes(4).toString()
+    );
+    this.messaging = getMessaging(firebase);
+  }
+
+  async sendMessage(
+    options: IPushOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    delete (options.overrides as any).notificationIdentifiers;
```

Comment:
Hey @scopsy @raikasdev 
what is the purpose of this one?
Looks like the test 'options.overrides' is undefined which causes the test to fails.

Reply:
```suggestion
    delete (options.overrides as any)?.notificationIdentifiers;
```
It should not pass the identifiers from the overrides when sending a message. I think you can just do it like this.

Reply:
Awesome thanks, I think there is more i am looking into it rn